### PR TITLE
Drop passfd module

### PR DESCRIPTION
**Pull request by luckyh**
* The `sendfd` function provided by the passfd module can be imitated by `socket.sendmsg`, for where python >= 3.3. Hence, we can safely drop this module to reduce the cost of maintainability.
* Drop the dependency of python headers as it is no longer needed after dropping the passfd module.
* Some minor fix of the rpm spec file

### CODE REVIEW

**Comment by luckyh** [comment]
@YongxueHong @clebergnu @richtja would you mind to review this PR? thanks!

**Comment by luckyh** [comment]
Hi @pevogam as the python3-devel dependency was dropped by this PR would you please help confirm it won't break anything to the avocado-i2n plugin? thanks!

File: virttest/qemu_monitor.py [hunk 1]
```diff
@@ -850,20 +849,28 @@ def _read_up_to_qemu_prompt(self, timeout=PROMPT_TIMEOUT):
                 pass
         return False, "\n".join(s.splitlines())
 
-    def _send(self, cmd):
+    def _send(self, cmd, fds=None):
         """
         Send a command without waiting for output.
 
         :param cmd: Command to send, type: bytes
```
Comment (pevogam) [inline virttest/qemu_monitor.py:856]: Can we add a description for the `fds` argument here? It would be consistent and add more understanding for the changes here.

File: virttest/qemu_monitor.py [hunk 2]
```diff
@@ -920,17 +927,15 @@ def cmd(self, cmd, timeout=CMD_TIMEOUT, debug=True, fd=None):
...
-            else:
-                # Send command
-                if debug:
-                    LOG.debug("Send command: %s" % cmd)
-                self._send(cmd.encode())
+            # Send command
+            msg = cmd.encode()
+            fds = (
+                [fd.fileno() if not isinstance(fd, int) else fd]
+                if fd is not None else None
```
Comment (pevogam) [inline virttest/qemu_monitor.py:934]: This can be used for inspiration to give a type and a one-sentence description for the file descriptors above.

File: virttest/qemu_monitor.py [hunk 3]
```diff
@@ -1846,15 +1851,23 @@ def _read_objects(self, timeout=READ_OBJECTS_TIMEOUT):
         self._events += [obj for obj in objs if "event" in obj]
         return objs
 
-    def _send(self, data):
+    def _send(self, data, fds=None):
         """
         Send raw bytes data without waiting for response.
 
         :param data: Data to send type: bytes
```
Comment (pevogam) [inline virttest/qemu_monitor.py:1858]: Same request here

**Review by pevogam — COMMENTED** [review]
Hi @luckyh, this is a great change as dropping dependencies is always welcome. I left a few comments and did some minimal runs that passed. I am doing a very large run with more potential corner cases right now but unless I write again consider the tests as passing.

**Comment by pevogam** [comment]
In the end I did get one somewhat fatal problem like
```
[stderr] Exception in thread 
[stderr] tail_thread_Dz7KhdwY_ssh  -o Us
[stderr] :
[stderr] Traceback (most recent call last):
[stderr]   File "/usr/lib64/python3.10/threading.py", line 1016, in _bootstrap_inner
[stderr]     
[stderr] DEBUG:aexpect.client:[192.168.1.1] Sending command: cat /var/log/messages
[stderr] self.run()
[stderr] 
[stderr]   File "/usr/lib64/python3.10/threading.py", line 953, in run
[stderr]     
[stderr] self._target(*self._args, **self._kwargs)
[stderr] 
[stderr]   File "/usr/lib/python3.10/site-packages/aexpect/client.py", line 671, in _tail
[stderr]     
[stderr] _print_line(line)
[stderr] 
[stderr]   File "/usr/lib/python3.10/site-packages/aexpect/client.py", line 633, in _print_line
[stderr]     
[stderr] self.output_func(*self.output_params + (text,))
[stderr] 
[stderr]   File "/usr/lib/python3.10/site-packages/virttest/utils_misc.py", line 428, in log_line
[stderr]     
[stderr] raise LogLockError("Could not acquire exclusive lock to access"
[stderr] 
[stderr] virttest.utils_misc
[stderr] .
[stderr] LogLockError
[stderr] : 
[stderr] Could not acquire exclusive lock to access _open_log_files
[stderr] 
```
but it is likely not related to the monitor sockets. @ldoktor Have you seen an error like this with aexpect ever before?

**Comment by ldoktor** [comment]
> In the end I did get one somewhat fatal problem like
> 
> ```
> [stderr] Exception in thread 
> [stderr] tail_thread_Dz7KhdwY_ssh  -o Us
> [stderr] :
> [stderr] Traceback (most recent call last):
> [stderr]   File "/usr/lib64/python3.10/threading.py", line 1016, in _bootstrap_inner
> [stderr]     
> [stderr] DEBUG:aexpect.client:[192.168.1.1] Sending command: cat /var/log/messages
> [stderr] self.run()
> [stderr] 
> [stderr]   File "/usr/lib64/python3.10/threading.py", line 953, in run
> [stderr]     
> [stderr] self._target(*self._args, **self._kwargs)
> [stderr] 
> [stderr]   File "/usr/lib/python3.10/site-packages/aexpect/client.py", line 671, in _tail
> [stderr]     
> [stderr] _print_line(line)
> [stderr] 
> [stderr]   File "/usr/lib/python3.10/site-packages/aexpect/client.py", line 633, in _print_line
> [stderr]     
> [stderr] self.output_func(*self.output_params + (text,))
> [stderr] 
> [stderr]   File "/usr/lib/python3.10/site-packages/virttest/utils_misc.py", line 428, in log_line
> [stderr]     
> [stderr] raise LogLockError("Could not acquire exclusive lock to access"
> [stderr] 
> [stderr] virttest.utils_misc
> [stderr] .
> [stderr] LogLockError
> [stderr] : 
> [stderr] Could not acquire exclusive lock to access _open_log_files
> [stderr] 
> ```
> 
> but it is likely not related to the monitor sockets. @ldoktor Have you seen an error like this with aexpect ever before?

Well, I had seen such messages, but it's been years since then. Anyway this exception has nothing to do with aexpect itself as aexpect is just executing avocado-vt's `log_line` function, which is failing to acquire a `threading.RLock` for some reason. If you have a clear reproducer I'd recommend running a debugger on it to see what processes are holding it.

File: virttest/qemu_monitor.py (hunk 1 above)
Comment (luckyh) [reply virttest/qemu_monitor.py]: Thanks, updated!

File: virttest/qemu_monitor.py (hunk 3 above)
Comment (luckyh) [reply virttest/qemu_monitor.py]: Thanks, updated!

**Comment by pevogam** [comment]
> > ```
> > 
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > but it is likely not related to the monitor sockets. @ldoktor Have you seen an error like this with aexpect ever before?
> 
> 
> Well, I had seen such messages, but it's been years since then. Anyway this exception has nothing to do with aexpect itself as aexpect is just executing avocado-vt's `log_line` function, which is failing to acquire a `threading.RLock` for some reason. If you have a clear reproducer I'd recommend running a debugger on it to see what processes are holding it.

So far what I know is that the problem only happens on aexpect session commands like `cat /var/log/messages` and reading the complete syslog. I have also confirmed that I see the issue beyond this pull request so the changes here are not related in any direct or indirect way.

**Review by pevogam — APPROVED** [review]
LGTM

**Review by YongxueHong — APPROVED** [review]
LGTM.
